lib/spack/spack/modules.py

```diff
@@ -261,6 +366,13 @@ class TclModule(EnvModule):
         UnsetEnv: 'unsetenv {name}\n'
     }
 
+    autoload_format = ('if ![ is-loaded {module_file} ] {{'
+                       '    puts stderr "Autoloading {module_file}"'
+                       '    module load {module_file}'
+                       '}}')
+
+    prerequisite_format = 'prereq {module_file}\n'
+
```

Comment:
@tgamblin Do you know if you can do something similar to `autoload_format` and `prerequisite_format` with dotkit?